app/src/main/java/at/roteskreuz/stopcorona/screens/debug/exposure_notifications/DebugExposureNotificationsFragment.kt

```diff
@@ -33,9 +35,31 @@ class DebugExposureNotificationsFragment : BaseFragment(R.layout.debug_contact_t
 
         exposureNotificationsSettingsButton.setOnClickListener { viewModel.jumpToSystemSettings() }
 
-        exposureNotificationsUploadTemporaryExposureKeysGreenButton.setOnClickListener { viewModel.uploadKeys(WarningType.REVOKE) }
-        exposureNotificationsUploadTemporaryExposureKeysRedButton.setOnClickListener { viewModel.uploadKeys(WarningType.RED) }
-        exposureNotificationsUploadTemporaryExposureKeysYellowButton.setOnClickListener { viewModel.uploadKeys(WarningType.YELLOW) }
+        val uploadKeylistener = View.OnClickListener(){button ->
+            val tan = exposureNotificationsTanEditText.text.toString()
+            if (tan.isNullOrBlank()){
+                activity?.let { Toast.makeText(activity,"please add TAN", Toast.LENGTH_SHORT)}
```

Comment:
Please don't use hardcoded strings

Reply:
also in the next line. Moreover, why are you using a toast instead of displaying the error in the editText?

Reply:
In debug screens I would allow some mess because they won't be part of the production release and the only purpose is to test something. But you're right, the mess doesn't have to be too obvious 🙂